plugins/syncbeacon/plugin.go

```diff
@@ -0,0 +1,217 @@
+	beaconSyncMap = make(map[string]beaconSync, len(beaconNodesPublicKeys))
+	for _, pubKey := range beaconNodesPublicKeys {
+		beaconSyncMap[pubKey] = beaconSync{
+			payload: nil,
+			synced:  false,
+		}
+	}
+
+	messagelayer.Tangle().Events.MessageSolid.Attach(events.NewClosure(func(cachedMessage *message.CachedMessage, cachedMessageMetadata *tangle.CachedMessageMetadata) {
+		cachedMessageMetadata.Release()
+		cachedMessage.Consume(func(msg *message.Message) {
+			payload := &Payload{}
+			err := payload.Unmarshal(msg.Payload().Bytes())
+			if err != nil {
+				return
+			}
+			if !IsSyncBeaconPayload(payload) {
+				return
+			}
+
+			handlePayload(payload, msg.IssuerPublicKey())
+		})
+	}))
+}
+
+// handlePayload handles the received payload. It does the following checks:
+// It checks that the issuer of the payload is a followed node.
+// The time that payload was sent is not greater than  CfgSyncBeaconMaxTimeWindowSec. If the duration is longer than CfgSyncBeaconMaxTimeWindowSec, we consider that beacon to be out of sync till we receive a newer payload.
+// More than syncPercentage of followed nodes are also synced, the node is set to synced. Otherwise, its set as desynced.
+func handlePayload(syncBeaconPayload *Payload, issuerPublicKey ed25519.PublicKey) {
+	issuerPublicKeyStr := issuerPublicKey.String()
+	//check if issuer is in configured beacon follow list
+	shouldAccept := false
+	for _, pubKey := range beaconNodesPublicKeys {
+		if pubKey == issuerPublicKeyStr {
+			shouldAccept = true
+			break
+		}
+	}
+	if !shouldAccept {
+		return
+	}
+	_beaconSync := beaconSync{
+		payload: syncBeaconPayload,
+		synced:  true,
+	}
+
+	dur := time.Since(time.Unix(0, syncBeaconPayload.sentTime))
+	if dur.Seconds() > float64(config.Node().GetInt(CfgSyncBeaconMaxTimeWindowSec)) {
+		log.Infof("syncbeacon received from %s is too old", issuerPublicKey)
+		_beaconSync.synced = false
+	}
+
+	beaconSyncMap[issuerPublicKeyStr] = _beaconSync
+	updateSynced()
+}
+
+// updateSynced checks the beacon nodes and update the nodes sync status
+func updateSynced() {
+	beaconNodesSyncedCount := 0.0
+	for _, beaconSync := range beaconSyncMap {
+		if beaconSync.synced {
+			beaconNodesSyncedCount++
+		}
+	}
+	synced := true
+	if len(beaconSyncMap) > 0 {
+		synced = beaconNodesSyncedCount/float64(len(beaconSyncMap)) >= syncPercentage
+	}
+
+	if synced {
+		sync.MarkSynced()
+		log.Info("synced from beacon")
+	} else {
+		sync.MarkDesynced()
+	}
+}
+
+// broadcastSyncBeaconPayload broadcasts sends a sync beacon to the message layer
+func broadcastSyncBeaconPayload() {
+	syncBeaconPayload := NewSyncBeaconPayload(time.Now().UnixNano())
+	messagelayer.MessageFactory().IssuePayload(syncBeaconPayload)
+	log.Info("issued sync beacon")
+}
+
+// cleanupFollowNodes cleans up offline nodes by setting their sync status to false after a configurable time window.
+func cleanupFollowNodes() {
+	if len(beaconNodesPublicKeys) == 0 {
```

Comment:
The background worker should not even be started in `run` if there is no beacon nodes configured. 
Similarly, the plugin should not attach to the message tangle if there's no beacon nodes configured. 

If you split the plugin this is a configuration error, i.e. if the plugin is enabled and then there should always be at least 1 beacon node configured for the sync checking part. So this could maybe even panic.